python/cudf/cudf/core/multiindex.py

```diff
@@ -1302,6 +1302,54 @@ def _poplevels(self, level):
 
         return popped
 
+    @_cudf_nvtx_annotate
+    def swapij(self, i=0, j=1):
+        """
+        Swap level i with level j.
+        Calling this method does not change the ordering of the values.
+
+        Parameters
+        ----------
+        i : int, str, default 0
+            First level of index to be swapped.
+        j : int, str, default 1
+            Second level of index to be swapped.
+
+        Returns
+        -------
+        MultiIndex
+            A new MultiIndex.
+
+
+        Examples
+        --------
+        >>> import cudf
+        >>> mi = cudf.MultiIndex(levels=[['a', 'b'], ['bb', 'aa']],
+        ...                    codes=[[0, 0, 1, 1], [0, 1, 0, 1]])
+        >>> mi
+        MultiIndex([('a', 'bb'),
+            ('a', 'aa'),
+            ('b', 'bb'),
+            ('b', 'aa')],
+           )
+        >>> mi.swapij(0, 1)
+        MultiIndex([('bb', 'a'),
+            ('aa', 'a'),
+            ('bb', 'b'),
+            ('aa', 'b')],
+           )
+        """
+        result = self.copy()
+        new_levels = result.levels
+        new_codes = result.codes
+        new_names = list(result.names)
+
+        new_levels[i], new_levels[j] = new_levels[j], new_levels[i]
+        new_codes[i], new_codes[j] = new_codes[j], new_codes[i]
```

Comment:
I am a little surprised that this works. `i` and `j` here are indices, but codes is a `DataFrame` that needs to be indexed by column names, not numbers. Are you sure that this is behaving the way that you expect? What happens if you replace `i` and `j` with `result.names[i]` and `result.names[j]` (which is what I would have expected you to have to do)?